### PR TITLE
feat: Basic comments support

### LINKED ISSUE #11: As an authenticated user I need to be able to comment on each entry
Authenticated users should be able to comment, report on each entries

### PR DESCRIPTION
**Pull request by umstek**
Adds support for multi-level comments/replies. 

Closes #11 if merged. 

Known issues: 

- Architecture needs review
- UI not finalized
- Authentication not added

### CODE REVIEW

File: server/src/modals/entry.js [hunk 1]
```diff
@@ -32,6 +32,27 @@ const locationSchema = mongoose.Schema(
...
+    id: {
+      type: String,
+      required: true,
+    },
+    author: authorSchema,
+    body: {
+      type: String,
+      required: true,
+    },
+    // @todo: do we track only the number of likes (store own likes)?
```
Comment (thinkholic) [inline server/src/modals/entry.js:46]: Do we need likes?

Comment (umstek) [reply server/src/modals/entry.js:46]: Haven't added, but it's in the design

**Comment by umstek** [comment]
Fixed the issues happened after integrating upstream changes

**Comment by thinkholic** [comment]
@umstek Nice work. Notice one issue as comments won't reflect unless page reloading again.

CommentsTree feature is a nice addition. :)

**Comment by umstek** [comment]
> @umstek Nice work. Notice one issue as comments won't reflect unless page reloading again.
> 
> CommentsTree feature is a nice addition. :)

Nested comments feature was in the design. There are several ways to do this but I hope this is enough for this project.

I can fix the comments updating issue. 
Are we using any state management library/architecture?

**Comment by thinkholic** [comment]
Let’s simply use Context api only. No need any third party state management
libraries at this point IMO. Btw, we are a bit out from the designs now for
the sake of getting the MVP done asap. So we can use our beat judgement

**Comment by umstek** [comment]
> @umstek Nice work. Notice one issue as comments won't reflect unless page reloading again.

@thinkholic Fixed the above issue.

**Review by thinkholic — APPROVED** [review]

**Comment by thinkholic** [comment]
Approved and merged. @umstek It would be perfect we can enable delete their own comment feature? wdyt?

**Comment by umstek** [comment]
> Approved and merged. @umstek It would be perfect we can enable delete their own comment feature? wdyt?

Will do. Have already added the endpoint.

**Pull request merged**